Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -106,5 +465,33 @@ namespace ScriptAutomation
 
         behaviorContext->Method("IdleFrames", &Bindings::IdleFrames);
         behaviorContext->Method("IdleSeconds", &Bindings::IdleSeconds);
+        behaviorContext->Method("ResizeViewport", &Bindings::ResizeViewport);
+
+        behaviorContext->Method("ResolvePath", &Bindings::ResolvePath);
+        behaviorContext->Method("NormalizePath", &Bindings::NormalizePath);
+        behaviorContext->Method("DegToRad", &Bindings::DegToRad);
+        behaviorContext->Method("GetRenderApiName", &Bindings::GetRenderApiName);
+        behaviorContext->Method("GetScreenshotOutputPath", &Bindings::GetScreenshotOutputPath);
+        behaviorContext->Method("GetProfilingOutputPath", &Bindings::GetProfilingOutputPath);
+
+        // Screenshots...
+        behaviorContext->Method("CaptureScreenshot", &Bindings::CaptureScreenshot);
+        behaviorContext->Method("CaptureScreenshotWithPreview", &Bindings::CaptureScreenshotWithPreview);
+        behaviorContext->Method("CapturePassAttachment", &Bindings::CapturePassAttachment);
+
+        // Profiling data...
+        behaviorContext->Method("CapturePassTimestamp", &Bindings::CapturePassTimestamp);
+        behaviorContext->Method("CaptureCpuFrameTime", &Bindings::CaptureCpuFrameTime);
+        behaviorContext->Method("CapturePassPipelineStatistics", &Bindings::CapturePassPipelineStatistics);
+        behaviorContext->Method("CaptureCpuProfilingStatistics", &Bindings::CaptureCpuProfilingStatistics);
+        behaviorContext->Method("CaptureBenchmarkMetadata", &Bindings::CaptureBenchmarkMetadata);
+
+        // Settings Registery query
+        behaviorContext->Method("SettingsRegistryKeyExists", &Bindings::SettingsRegistry_QueryKeyExists);
+        behaviorContext->Method("SettingsRegistryGetBool",   &Bindings::SettingsRegistry_GetBool);
+        behaviorContext->Method("SettingsRegistryGetSInt",   &Bindings::SettingsRegistry_GetSInt);
+        behaviorContext->Method("SettingsRegistryGetUInt",   &Bindings::SettingsRegistry_GetUInt);
+        behaviorContext->Method("SettingsRegistryGetFloat",  &Bindings::SettingsRegistry_GetFloat);
+        behaviorContext->Method("SettingsRegistryGetString", &Bindings::SettingsRegistry_GetString);
```

Comment:
IIRC the behavior context system supports reflection of templated functions with automatic type suffixes. (I followed the same naming when implementing the lua material functor API). To be consistent with these suffixes, I recommend these names:
```suggestion
        behaviorContext->Method("SettingsRegistryKeyExists", &Bindings::SettingsRegistry_QueryKeyExists);
        behaviorContext->Method("SettingsRegistryGet_bool",   &Bindings::SettingsRegistry_GetBool);
        behaviorContext->Method("SettingsRegistryGet_int",   &Bindings::SettingsRegistry_GetSInt);
        behaviorContext->Method("SettingsRegistryGet_uint",   &Bindings::SettingsRegistry_GetUInt);
        behaviorContext->Method("SettingsRegistryGet_float",  &Bindings::SettingsRegistry_GetFloat);
        behaviorContext->Method("SettingsRegistryGet_string", &Bindings::SettingsRegistry_GetString);
```

Reply:
I'll look into this.

Reply:
> IIRC the behavior context system supports reflection of templated functions with automatic type suffixes. (I followed the same naming when implementing the lua material functor API). To be consistent with these suffixes, I recommend these names:

I'll look into this.

Reply:
The SettingsRegistry has already been reflected to the [BehaviorContext](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryScriptUtils.cpp#L320-L335)

Reply:
This is not the same context

Reply:
Yeah, but you could just pass this BehaviorContext instance into the [ReflectSettingsRegistryToBehaviorContext](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryScriptUtils.h#L24) function

Reply:
We already have 2 SettingsRegistry lua query interfaces, this would add a 3rd :)

Reply:
This would actually reduce the Settings Registry lua interfaces down to the 1, since the original one works with any kind of settings registry.

Reply:
`local frameCaptureCount = g_SettingsRegistry:GetUInt(FrameCaptureCountRegistryKey):value_or(DEFAULT_FRAME_COUNT)`

vs

`local frameCaptureCount = SettingsRegistryGetUInt(FrameCaptureCountRegistryKey):value_or(DEFAULT_FRAME_COUNT)`

Reply:
That looks the same to me.

Either way, we shouldn't be defining the SettingsRegistry scripting bindings in two different files, so at the very least the bindings in ScriptAutomationScriptingBindings.cpp should be in the same file as the other bindings.

Reply:
I do recall the example.